cl/assets/react/_disclosure_helpers.tsx

```diff
@@ -1,256 +0,0 @@
-  if (child == null) {
-    return false;
-  }
-
-  let node = child.parentNode;
-  while (node != null) {
-    if (node == parent) {
-      return true;
-    }
-    node = node.parentNode;
-  }
-  return false;
-}
-
-export const fetchDisclosure = async (setData: any, parameters: any) => {
-  try {
-    const response: boolean | Query = await appFetch(
-      `/api/rest/v4/financial-disclosures/?person=${parameters['judge_id']}&id=${parameters['disclosure_id']}`
-    );
-    if (typeof response != 'boolean') {
-      const result = response['results'][0];
-      setData(result);
-    }
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-export const convertTD = (value: string | number, table: string, key: string) => {
-  if (value == -1 || !value) {
-    return '';
-  }
-  if (['Investments', 'Debts'].indexOf(table) == -1) {
-    return value;
-  }
-  if (['transaction_value_code', 'value_code', 'gross_value_code'].indexOf(key) > -1) {
-    return `$${GROSS_VALUE[value]}\u00a0(${value})`;
-  }
-  if (['income_during_reporting_period_code', 'transaction_gain_code'].indexOf(key) > -1) {
-    return `$${INCOME_GAIN[value]}\u00a0(${value})`;
-  }
-  if (key == 'gross_value_method' && value) {
-    return `${VALUATION_METHODS[value]}\u00a0(${value})`;
-  }
-  return value;
-};
-
-const fetchData = async (query: string, setData: (arg0: Row[]) => void) => {
-  try {
-    const response: boolean | Query = await appFetch(
-      `/api/rest/v4/disclosure-typeahead/?fullname=${query}&page_size=5`
-    );
-    if (typeof response != 'boolean') {
-      const results: Row[] = response['results'];
-      setData(results);
-    }
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-const InstantSearchResults = (small: boolean) => {
-  let size: string;
-  if (small) {
-    size = 'input-md form-control';
-  } else {
-    size = 'input-lg form-control';
-  }
-
-  const [data, setData] = React.useState<Row[]>([]);
-  const [visible, setVisible] = React.useState<boolean>(false);
-  const [query, setQuery] = React.useState<string>('');
-
-  const handleClickOutside = (event: Event) => {
-    const query_container = document.getElementById('main-query-box');
-    const child: HTMLElement = event.target as HTMLInputElement;
-    if (isDescendant(query_container, child)) {
-      setVisible(true);
-    } else {
-      setVisible(false);
-    }
-  };
-
-  const handleEsc = (event: { keyCode: number }) => {
-    if (event.keyCode === 27) {
-      setVisible(false);
-    }
-  };
-
-  React.useEffect(() => {
-    document.addEventListener('click', handleClickOutside, true);
-    window.addEventListener('keydown', handleEsc);
-  }, []);
-
-  const debounceFetchJudge = React.useMemo(() => debounce(fetchData, 300), []);
-
-  function update({ ...data }) {
-    //Trim whitespace to require two non whitespace characters.
-    const query: string = data.target.value.replace(/(^\s+|\s+$)/g, '');
```

Comment:
Is this getting done still?

Reply:
Yes! HTMX handles this with `hx-trigger="keyup changed delay:300ms"` on the input element. See line 50 in `financial_disclosures_home.html`.